lib/chef/provider/package/yum.rb

```diff
@@ -1064,6 +1068,9 @@ def load_current_resource
             parse_arch
           end
 
+          # normalize internal representation as an array
+          @new_resource.package_name(as_array(@new_resource.package_name))
```

Comment:
as_array is a generalized helper so I didn't have to add a helper for every possible thing. In this case, `package_name_array` operates on `new_resource`, but we need `@new_resource`, it's not clear to me these are the same and this is the only instance like that so I didn't make yet another helper/